src/models/HoldRequestConsumerModel.js

```diff
@@ -26,6 +28,50 @@ function HoldRequestConsumerModel(records) {
   this.setRecords = (records) => {
     this.records = records;
   };
+
+  this.getItemServiceApiUrls = () => {
+    return this.itemApiUrls;
+  };
+
+  this.setItemServiceApiUrls = (urlsArray) => {
+    this.itemApiUrls = urlsArray;
+  };
+
+  this.mergeRecordsBySourceAndRecordId = (initialRecords, updatedRecords) => {
+    const functionName = 'mergeRecordsBySourceAndRecordId';
+
+    if (!initialRecords.length) {
```

Comment:
I made a suggestion above to check for all conditions in one if  and then return the Promise rejection.